api/app/controllers/spree/api/address_book_controller.rb

```diff
@@ -0,0 +1,43 @@
+module Spree
+  module Api
+    class AddressBookController < Spree::Api::BaseController
+      # Note: the AddressBook is the resource to think about here, not individual addresses
+
+      before_filter :ensure_api_token
+
+      def mine
+         @user_addresses = current_api_user.user_addresses
+      end
+
+      def update
+        address_params = address_book_params
+        default_flag = address_params.delete(:default)
+        address = current_api_user.save_in_address_book(address_params, default_flag)
+        if address.valid?
+          @user_addresses = current_api_user.user_addresses
+          render :mine, status: :ok
```

Comment:
We understand Law of Demeter quite differently, but nevertheless, I developed this API at the same time as a UI that uses it (which I hope we can generify and share with the community). In every case we wanted to render the updated AddressBook after every modification. I still can't see a 303 status code and extra round-trip as an improvement. I also checked the solidus code and see no use of a 303 or a :see_other status code, so it's not a convention already in place that this code goes against.